mm2src/mm2_main/src/lr_swap/lr_quote.rs

```diff
@@ -0,0 +1,1078 @@
+                .await
+                .map_mm_err()?;
+            wei_to_coins_mm_number(atomic_swap_taker_amount, taker_coin.decimals()).map_mm_err()?
+        };
+        log::debug!(
+            "check_order_limits: {}/{} atomic_swap_taker_amount={} max_taker_vol={} min_taker_vol={}",
+            candidate.maker_order.maker_ticker(),
+            candidate.maker_order.taker_ticker(),
+            atomic_swap_taker_amount.to_decimal(),
+            candidate.maker_order.max_taker_vol().decimal,
+            candidate.maker_order.min_taker_vol().decimal,
+        );
+        if atomic_swap_taker_amount.to_ratio() > candidate.maker_order.max_taker_vol().rational
+            || atomic_swap_taker_amount.to_ratio() < candidate.maker_order.min_taker_vol().rational
+        {
+            log::debug!(
+                "check_order_limits: {}/{} out of order min/max, skipping candidate",
+                candidate.maker_order.maker_ticker(),
+                candidate.maker_order.taker_ticker()
+            );
+            return Ok(false);
+        }
+
+        let atomic_swap_maker_amount = if let Some(ref lr_data_1) = candidate.lr_data_1 {
+            wei_to_coins_mm_number(lr_data_1.src_amount()?, lr_data_1.src_decimals()?).map_mm_err()?
+        } else {
+            let atomic_swap_maker_amount = candidate
+                .atomic_swap_maker_amount
+                .ok_or(LrSwapError::InternalError("no atomic swap maker amount".to_owned()))?;
+            let maker_coin = lp_coinfind_or_err(ctx, &candidate.maker_order.maker_ticker())
+                .await
+                .map_mm_err()?;
+            wei_to_coins_mm_number(atomic_swap_maker_amount, maker_coin.decimals()).map_mm_err()?
+        };
+        log::debug!(
+            "check_order_limits: {}/{} atomic_swap_maker_amount={} max_maker_vol={} min_maker_vol={}",
+            candidate.maker_order.maker_ticker(),
+            candidate.maker_order.taker_ticker(),
+            atomic_swap_maker_amount.to_decimal(),
+            candidate.maker_order.max_maker_vol().decimal,
+            candidate.maker_order.min_maker_vol().decimal,
+        );
+        if atomic_swap_maker_amount.to_ratio() > candidate.maker_order.max_maker_vol().rational
+            || atomic_swap_maker_amount.to_ratio() < candidate.maker_order.min_maker_vol().rational
+        {
+            log::debug!(
+                "check_order_limits: {}/{} out of order min/max, skipping candidate",
+                candidate.maker_order.maker_ticker(),
+                candidate.maker_order.taker_ticker()
+            );
+            return Ok(false);
+        }
+        Ok(true)
+    }
+
+    /// Select the best swap path, by minimum of total swap price, including LR steps and atomic swap)
+    #[allow(clippy::result_large_err)]
+    async fn select_best_swap(self, ctx: &MmArc) -> MmResult<(LrSwapCandidateInfo, MmNumber), LrSwapError> {
```

Comment:
Do all 1inch swaps use the same gas amount? I see that the ClassicSwapData fetched from the 1inch API provides an estimated gas cost, and this should be factored in, as a swap with a better price might have a higher gas cost, making it more expensive overall